src/_components/_interactive/button/button.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { FunctionComponent, MouseEventHandler } from 'react';
+
+import { Icon } from '../../display/icon/icon';
+import { IconName } from '../../display/icon/icon.types';
+
+import { buttonMap, ButtonTheme, ButtonType } from './button.types';
+
+interface ButtonProps {
+  label: string;
+  id?: string;
+  iconName?: IconName;
```

Comment:
It's probably worth having a way to swap which side of the text the icon sits - see sign-out vs sign-in on the Figma.